setup-bootstrap.sh

```diff
@@ -1,22 +1,63 @@
-#! /bin/bash
+#!/usr/bin/env bash
+# Copyright (c) 2022, Linaro. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
 
-set -e
+# Exit on command failure
+set -o errexit
+
+# Fail on unset variable
+# Use "${VARNAME-}" instead of "$VARNAME" to access unset variable(s)
+set -o nounset
+
+# Enable debug mode if $TRACE is set
+# To enable, run with: "TRACE=1 ./script.sh"
+if [[ "${TRACE-0}" == "1" ]]; then
+    set -o xtrace
+fi
+
+# Check if the first arg is -h or --help
+if [[ "${1-}" =~ ^-*h(elp)?$ ]]; then
+    echo "Usage: ./setup-bootstrap.sh
+
+Generates a device class certicate and keypair that enables client devices to
+connect to the bootstrap server during device provisioning, or to interact with
+the REST API using mutual TLS authentication.
+
+This certificate and private key must be available on client devices. If the
+certificate and matching private key is not available on the client device, the
+server will refuse the TLS connection.
+
+The following files are placed them in the 'certs' folder:
+
+- certs/BOOTSTRAP.crt      Client certificate for the bootstrap server
+- certs/BOOTSTRAP.key      Private key for the bootstrap server client
+- certs/bootstrap_crt.txt  A C string copy of BOOTSTRAP.crt (as a convenience)
+- certs/bootstrap_key.txt  A C string copy of BOOTSTRAP.key (as a convenience)
+
+You can view the content of the certificate via:
+
+   $ openssl x509 -in certs/BOOTSTRAP.crt -noout -text
+"
+    exit
+fi
 
 # Generate a keypair for initial device provisioning.  The idea is
 # that this private key and certificate can be programmed into the
 # devices, within a class, at factory time.  This will allow these
 # devices to request certificates later to authenticate themselves.
 
 # For now, write them to a single name
-if [ -f certs/BOOTSTRAP.crt -o -f certs/BOOTSTRAP.key ];
+if [ -f certs/BOOTSTRAP.crt ] || [ -f certs/BOOTSTRAP.key ];
```

Comment:
Any particular reason for this change? We're already relying on bash, and the problems with -o stem from posix defining `[` to only take 4 arguments.

Reply:
Just blindly following shellcheck out of habit, but you're right that we are explicitly setting bash here.